### PR TITLE
Add "Allow resharing" option to share dialog

### PR DESCRIPTION
**Pull request by claucambra**
Screenshot 2023-08-10 at 16 15 23

### CODE REVIEW

File: src/gui/sharepermissions.h [hunk 1]
```diff
@@ -23,12 +23,12 @@ namespace OCC {
  * Possible permissions, must match the server permission constants
  */
 enum SharePermission {
-    SharePermissionRead     = 1 << 0,
-    SharePermissionUpdate   = 1 << 1,
-    SharePermissionCreate   = 1 << 2,
-    SharePermissionDelete   = 1 << 3,
-    SharePermissionShare    = 1 << 4,
-    SharePermissionDefault  = 1 << 30
```
Comment (allexzander) [inline src/gui/sharepermissions.h:31]: @claucambra Could you elaborate? I am not sure I understand the change

**Review by allexzander — CHANGES_REQUESTED** [review]
@claucambra one change please, and one question from me

File: src/gui/sharepermissions.h (hunk 1 above)
Comment (claucambra) [reply src/gui/sharepermissions.h:31]: `SharePermissionDefault` is not really a default value but rather an "all" value, hence it having the value 31. in a bit field this is 11111 . The previous value was a gigantic int that was not 31

File: src/gui/filedetails/sharemodel.cpp [hunk 2]
```diff
@@ -1306,6 +1311,12 @@ bool ShareModel::canShare() const
     return _maxSharingPermissions & SharePermissionShare;
 }
 
+bool ShareModel::resharingAllowed() const
```
[the remark this answers is not in the document]
Comment (claucambra) [reply src/gui/filedetails/sharemodel.cpp]: Good point, have renamed to `serveAllowsResharing`

**Review by allexzander — APPROVED** [review]
@claucambra 
Kindly squash commit history

**Pull request merged**